File: Easy/83_Remove_Duplicates_from_Sorted_List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        if(head is None):
            return None
        current = head

        while(current.next is not None):
            if(current.val == current.next.val):
                current.next = current.next.next
            else:
                current = current.next
        return head

        # My first try using after but it wasnt nessissary
        # if(head is None):
        #     return None
        # current = head
        # after = head.next
        # while(after is not None):
        #     if(current.val == after.val):
        #         current.next, after = after.next, after.next
        #     else:
        #         current = after
        #         after = after.next
        # return head


    
        # Best solution on LC
        # if head==None:
        #     return
        # f=head
        # while f and f.next:
        #     if f.val==f.next.val:
        #         f.next=f.next.next
        #         continue
        #     f=f.next    
            
        # return head  

File: EasyJS/83_Remove_Duplicates_from_Sorted_List.js
var deleteDuplicates = function (head) {
  let headcpy = head;
  if (head == null) {
    return head;
  }
  if (head.next == null) {
    return head;
  }
  let left = head;
  let right = head.next;
  while (right != null) {
    if (left.val == right.val) {
      left.next = right.next;
      right = left.next;
    } else {
      left = left.next;
      right = right.next;
    }
  }
  return headcpy;
};
